### PR TITLE
add optional security headers as comment to .htaccess

### PR DESCRIPTION
**Pull request by Rotzbua**
Hi,
I added the most used optional header to .htaccess as a comment, so it do not have any effect.
I think it is helpful for people who like to harden their installation.

Source: https://www.owasp.org/index.php/List_of_useful_HTTP_headers
Testing: https://securityheaders.io/

Best

### CODE REVIEW

**Comment by sebix** [comment]
HSTS and HPKP should be done per-domain (system-wide), not per-application.

I'd like to see the others enabled by default, as they totally make sense on application basis (to prevent attacks on users of the application). But I'm not sure if they should be set via PHP or `.htaccess`. Setting them via PHP enables them for all webservers, not only Apache.

**Comment by Rotzbua** [comment]
I agree that HSTS should be done by system config and per (sub)domain. But shared hosters only allow custom headers by .htaccess or php.

I agree with your .htaccess only for Apache argument, but then the other comment for CSP (Content-Security-Policy) should be removed.

File: .htaccess [hunk 1]
```diff
@@ -46,7 +46,34 @@ SetOutputFilter DEFLATE
...
+#Header set X-Frame-Options SAMEORIGIN
+
+# X-Content-Type-Options
+# It prevents Google Chrome and Internet Explorer from trying to mime-sniff the content-type of a response away from the one being declared by the server.
+#Header set X-Content-Type-Options: "nosniff"
+
+# CSP - Content Security Policy
 # for better privacy/security ask browsers to not set the Referer
+# more flags for script, stylesheets and images available, read RFC for more information
 #Header set Content-Security-Policy "referrer no-referrer"
```
Comment (sebix) [inline .htaccess]: I can't find `no-referrer` in the RFC: https://www.w3.org/TR/CSP/

Comment (alecpl) [inline .htaccess]: https://w3c.github.io/webappsec-referrer-policy/ It's a working draft, but some browsers support it. https://developer.mozilla.org/en/docs/Web/Security/CSP/CSP_policy_directives

**Pull request merged**